src/test_l3_dsAudio.c

```diff
@@ -0,0 +1,3045 @@
+  {  NULL, -1 }
+};
+
+/* dsAudioARCTypes_t */
+const static ut_control_keyStringMapping_t dsAudioARCTypes_mapTable [] = {
+  { "dsAUDIOARCSUPPORT_NONE", (int32_t)dsAUDIOARCSUPPORT_NONE },
+  { "dsAUDIOARCSUPPORT_ARC",  (int32_t)dsAUDIOARCSUPPORT_ARC  },
+  { "dsAUDIOARCSUPPORT_eARC", (int32_t)dsAUDIOARCSUPPORT_eARC },
+  {  NULL, -1 }
+};
+
+/* dsAudioStereoMode_t */
+const static ut_control_keyStringMapping_t dsAudioStereoMode_mapTable [] = {
+  { "dsAUDIO_STEREO_UNKNOWN",  (int32_t)dsAUDIO_STEREO_UNKNOWN  },
+  { "dsAUDIO_STEREO_MONO",     (int32_t)dsAUDIO_STEREO_MONO     },
+  { "dsAUDIO_STEREO_STEREO",   (int32_t)dsAUDIO_STEREO_STEREO   },
+  { "dsAUDIO_STEREO_SURROUND", (int32_t)dsAUDIO_STEREO_SURROUND },
+  { "dsAUDIO_STEREO_PASSTHRU", (int32_t)dsAUDIO_STEREO_PASSTHRU },
+  { "dsAUDIO_STEREO_DD",       (int32_t)dsAUDIO_STEREO_DD       },
+  { "dsAUDIO_STEREO_DDPLUS",   (int32_t)dsAUDIO_STEREO_DDPLUS   },
+  { "dsAUDIO_STEREO_MAX",      (int32_t)dsAUDIO_STEREO_MAX      },
+  {  NULL, -1 }
+};
+
+/* dsAudioInput_t */
+const static ut_control_keyStringMapping_t dsAudioInput_mapTable [] = {
+  { "dsAUDIO_INPUT_PRIMARY", (int32_t)dsAUDIO_INPUT_PRIMARY },
+  { "dsAUDIO_INPUT_SYSTEM",  (int32_t)dsAUDIO_INPUT_SYSTEM  },
+  { "dsAUDIO_INPUT_MAX",     (int32_t)dsAUDIO_INPUT_MAX     },
+  {  NULL, -1 }
+};
+
+/* dsError_t */
+const static ut_control_keyStringMapping_t dsError_mapTable [] = {
+  { "dsERR_NONE",                    (int32_t)dsERR_NONE                    },
+  { "dsERR_GENERAL",                 (int32_t)dsERR_GENERAL                 },
+  { "dsERR_INVALID_PARAM",           (int32_t)dsERR_INVALID_PARAM           },
+  { "dsERR_INVALID_STATE",           (int32_t)dsERR_INVALID_STATE           },
+  { "dsERR_ALREADY_INITIALIZED",     (int32_t)dsERR_ALREADY_INITIALIZED     },
+  { "dsERR_NOT_INITIALIZED",         (int32_t)dsERR_NOT_INITIALIZED         },
+  { "dsERR_OPERATION_NOT_SUPPORTED", (int32_t)dsERR_OPERATION_NOT_SUPPORTED },
+  { "dsERR_RESOURCE_NOT_AVAILABLE",  (int32_t)dsERR_RESOURCE_NOT_AVAILABLE  },
+  { "dsERR_OPERATION_FAILED",        (int32_t)dsERR_OPERATION_FAILED        },
+  { "dsErr_MAX",                     (int32_t)dsErr_MAX                     },
+  {  NULL, -1 }
+};
+
+/* bool */
+const static ut_control_keyStringMapping_t bool_mapTable [] = {
+  { "false", (int32_t)false },
+  { "true",  (int32_t)true  },
+  {  NULL, -1 }
+};
+
+/**
+ * @brief This function clears the stdin buffer.
+ *
+ * This function clears the stdin buffer.
+ */
+void readAndDiscardRestOfLine(FILE* in)
+{
+   int c;
+   while ( (c = fgetc(in)) != EOF && c != '\n');
+}
+
+/**
+ * @brief Callback function for Headphone connection status.
+ *
+ * This function is invoked whenever a change occurs in the Headphone connection.
+ */
+static void audioOutPortConnectCB(dsAudioPortType_t portType, uint32_t uiPortNo, bool isPortCon)
+{
+    UT_LOG_INFO("Received Connection status callback port: %s, port number: %d, Connection: %s",
```

Comment:
we can try printing this to the terminal, but I think the the future you may want to specify via the menu an optional log filename that you write too.

don't need to change it now, but something for consideration